sdjournal/journal.go

```diff
@@ -1118,3 +1136,30 @@ func (j *Journal) GetCatalog() (string, error) {
 
 	return catalog, nil
 }
+
+// GetBootID get systemd boot id
+func (j *Journal) GetBootID() (string, error) {
+	sd_id128_get_boot, err := getFunction("sd_id128_get_boot")
+	if err != nil {
+		return "", err
+	}
+
+	var boot_id C.sd_id128_t
+	r := C.my_sd_id128_get_boot(sd_id128_get_boot, &boot_id)
+	if r < 0 {
+		return "", fmt.Errorf("failed to get boot id: %s", syscall.Errno(-r).Error())
+	}
+
+	sd_id128_to_string, err := getFunction("sd_id128_to_string")
+	if err != nil {
+		return "", err
+	}
+
+	c := (*C.char)(C.malloc(33))
+	defer C.free(unsafe.Pointer(c))
+	C.my_sd_id128_to_string(sd_id128_to_string, boot_id, c)
```

Comment:
This is discarding the return value and thus possibly ignoring error values. It should capture `r` and check it (like in the other C call above).

Reply:
my_sd_id128_to_string return is char, so I'm going to deal with possible errors by checking the bootid length.

Reply:
Sorry, my bad, I mixed it with `_from_string()`.